### PR TITLE
Firebase ML framework

### PR DESCRIPTION
**Pull request by ifielker**
Firebase ML framework, no implementation yet.

### CODE REVIEW

File: src/commands/ml-models-publish.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Command } from "../command";
+import { FirebaseModel } from "../ml/models";
+
+module.exports = new Command("ml:models:publish <modelId>")
+  .description("publishes the Firebase ML Model with the given modelId")
+  .action(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (bkendall) [inline src/commands/ml-models-publish.ts:7]: there is no any below

File: src/commands/ml-models-unpublish.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Command } from "../command";
+import { FirebaseModel } from "../ml/models";
+
+module.exports = new Command("ml:models:unpublish <modelId>")
+  .description("unpublishes the Firebase ML Model with the given modelId")
+  .action(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (bkendall) [inline src/commands/ml-models-unpublish.ts:7]: no any below

File: src/commands/ml-models-create.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Command } from "../command";
+import { FirebaseModel } from "../ml/models";
+
+module.exports = new Command("ml:models:create <displayName> [source]")
```
Comment (bkendall) [inline src/commands/ml-models-create.ts:4]:
```suggestion
export default new Command("ml:models:create <displayName> [source]")
```

`src/command.ts` is wired to expect default exports from TS files

Comment (samtstern) [reply src/commands/ml-models-create.ts:4]: Driving by ... is this true?  Many commands don't do this:
https://github.com/firebase/firebase-tools/blob/master/src/commands/emulators-start.ts#L6

File: src/commands/ml-models-unpublish.ts (hunk 2 above)
Comment (ifielker) [reply src/commands/ml-models-unpublish.ts:7]: done

File: src/commands/ml-models-create.ts (hunk 3 above)
Comment (ifielker) [reply src/commands/ml-models-create.ts:4]: done

File: src/commands/ml-models-publish.ts (hunk 1 above)
Comment (ifielker) [reply src/commands/ml-models-publish.ts:7]: done

File: src/commands/ml-models-create.ts (hunk 3 above)
Comment (bkendall) [reply src/commands/ml-models-create.ts]: I believe the majority do though. Functionally it's the same.

File: src/commands/ml-models-create.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { Command } from "../command";
+import { FirebaseModel } from "../ml/models";
+
+export default new Command("ml:models:create <displayName> [source]")
+  .description(
```
Comment (bkendall) [inline src/commands/ml-models-create.ts:5]: (I should add this to the contributing guide:)

`.description` should be a one-line description. `.help` can be used to provide more verbose text about how the command should be used.

Comment (ifielker) [reply src/commands/ml-models-create.ts]: Ok, I'll shorten the description and add the .help string later with the help of our docs person.

File: src/commands/ml-models-delete.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { Command } from "../command";
+
+module.exports = new Command("ml:models:delete <modelId>")
```
Comment (bkendall) [inline src/commands/ml-models-delete.ts:3]: prefer `export default` in this case
```suggestion
export default new Command("ml:models:delete <modelId>")
```

File: src/commands/ml-models-get.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { Command } from "../command";
+import { FirebaseModel } from "../ml/models";
+
+module.exports = new Command("ml:models:get <modelId>")
```
Comment (bkendall) [inline src/commands/ml-models-get.ts:4]: prefer `export default` in this case
```suggestion
export default new Command("ml:models:get <modelId>")
```

File: src/commands/ml-models-list.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { Command } from "../command";
+import { FirebaseModel } from "../ml/models";
+
+module.exports = new Command("ml:models:list")
```
Comment (bkendall) [inline src/commands/ml-models-list.ts:4]: prefer `export default` in this case
```suggestion
export default new Command("ml:models:list")
```

File: src/commands/ml-models-publish.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { Command } from "../command";
+import { FirebaseModel } from "../ml/models";
+
+module.exports = new Command("ml:models:publish <modelId>")
```
Comment (bkendall) [inline src/commands/ml-models-publish.ts:4]: prefer `export default` in this case
```suggestion
export default new Command("ml:models:publish <modelId>")
```

File: src/commands/ml-models-unpublish.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { Command } from "../command";
+import { FirebaseModel } from "../ml/models";
+
+module.exports = new Command("ml:models:unpublish <modelId>")
```
Comment (bkendall) [inline src/commands/ml-models-unpublish.ts:4]: prefer `export default` in this case
```suggestion
export default new Command("ml:models:unpublish <modelId>")
```

File: src/commands/ml-models-update.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import { Command } from "../command";
+import { FirebaseModel } from "../ml/models";
+
+module.exports = new Command("ml:models:update <modelId>")
```
Comment (bkendall) [inline src/commands/ml-models-update.ts:4]: prefer `export default` in this case
```suggestion
export default new Command("ml:models:update <modelId>")
```

**Pull request merged**